src/tenderly/spawn-devnet.ts

```diff
@@ -0,0 +1,44 @@
+/* eslint-disable @typescript-eslint/no-var-requires, no-console */
+import fs from "fs";
+import dotenv from "dotenv";
+import util from "util";
+import proc from "child_process";
+
+const asyncExec = util.promisify(proc.exec);
+
+dotenv.config();
+
+const {
+  TENDERLY_ACCESS_KEY,
+  TENDERLY_PROJECT_SLUG,
+  TENDERLY_ACCOUNT_ID,
+  TENDERLY_DEVNET_TEMPLATE,
+} = process.env;
+
+// eslint-disable-next-line max-len
+const command = `tenderly devnet spawn-rpc --project ${TENDERLY_PROJECT_SLUG} --template ${TENDERLY_DEVNET_TEMPLATE} --account ${TENDERLY_ACCOUNT_ID}  --access_key ${TENDERLY_ACCESS_KEY}`;
+
+
+const spawnDevNet = async () => {
+  const { stderr } = await asyncExec(command);
+  const devNetUrl = stderr.trim().toString();
+
+  console.log(`DEVNET_RPC_URL=${ devNetUrl }`);
+
+  // if file doesn't exist, create it
+  if (!fs.existsSync(".env")) {
+    fs.writeFileSync(".env", "");
+  }
+  const fileContent = fs.readFileSync(".env", "utf8");
+
+  const newFileContent = fileContent.replace(/DEVNET_RPC_URL=.*/g, "");
+  fs.writeFileSync(".env", newFileContent);
+  fs.appendFileSync(".env", `DEVNET_RPC_URL=${ devNetUrl }`);
```

Comment:
Can you explain this flow? 

We replace `DEVNET_RPC_URL=` in the file with an empty string`""`,
Then we write that empty string to the file
Then we append to that file with `DEVNET_RPC_URL=${devNetUrl}`

Why do we do it this way? Would not be simpler to see if `.env` contained a devNetUrl already and if so, replace it, or if it doesn't add it? This is presuming the URL is different each time as well.

Reply:
I actually lifted this piece of code :))
The way I see it, it's to also make sure the old link is removed before the new one added. Maybe for potential conflicts or to not fuse 2 values together. Not sure this can even happen, though...

"to see if .env contained a devNetUrl already and if so, replace it, or if it doesn't add it" - isn't that what is being done here though?

Reply:
I guess that is what it's doing, but it feels like you can just use a flag in `fs.writeFileSync` to overwrite contents of a file and then you can avoid doing this sort of round about check

https://nodejs.org/api/fs.html#file-system-flags